core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1534,19 +1538,25 @@ public void onReceive(Context context, Intent intent) {
         public void onReceive(Context context, Intent intent) {
             // sound is about to change, eg. bluetooth -> speaker
             Log.d(TAG, "Pausing playback because audio is becoming noisy");
-            pauseIfPauseOnDisconnect();
+            pauseIfPauseOnDisconnect(AudioOutputMode.PHONE_SPEAKER);
         }
         // android.media.AUDIO_BECOMING_NOISY
     };
 
     /**
      * Pauses playback if PREF_PAUSE_ON_HEADSET_DISCONNECT was set to true.
      */
-    private void pauseIfPauseOnDisconnect() {
-        if (UserPreferences.isPauseOnHeadsetDisconnect() && !isCasting()) {
-            if (mediaPlayer.getPlayerStatus() == PlayerStatus.PLAYING) {
-                transientPause = true;
-            }
+    private void pauseIfPauseOnDisconnect(AudioOutputMode outputMode) {
+        Log.d(TAG, "pauseIfPauseOnDisconnect() " + outputMode);
+        // If bluetooth is playing, headset is plugged in, headset unplugged, bluetooth should continue
+        // If headset is playing, bluetooth is connected, bluetooth is unplugged, headset should continue
+        // If headset or bluetooth is unplugged (the speaker should not be playing)
+        if (mediaPlayer.getPlayerStatus() == PlayerStatus.PLAYING) {
+            transientPause = true;
```

Comment:
Why does this need to be executed even when the player is not actually paused? I did not try to do it but I could imagine a scenario where the headset is unplugged and plugged in while playing with Bluetooth. It then has  `transientPause` set to `true`. When then pausing manually, the flag is still `true`. Wouldn't re-connecting then assume a wrong state?

Reply:
@ByteHamster  

Please go ahead and test it.   I tested your scenario and it works as expected (correctly)

`transientPause = true` is needed because `unpauseIfPauseOnDisconnect()` will set it to false later and it will be in the right state

Maybe the code could be done to be more understandable, let me know if you want me to document it better

Reply:
Sorry for the late reply. I can confirm that it works but I am still wondering why it is necessary to always set it to `true`. Is it because the Android system pauses playback automatically and we want it to resume? What would break if the assignment was moved down inside the `if` statement next to `mediaPlayer.pause`?

Reply:
@ByteHamster yup, your suggestion works.